Detectors/PHOS/simulation/include/PHOSSimulation/Detector.h

```diff
@@ -0,0 +1,179 @@
+  Bool_t ProcessHits(FairVolume* v = nullptr) final ;
+
+  ///
+  /// Add PHOS hit
+  /// Internally adding hits coming from the same track
+  ///
+  /// \param[in] trackID Index of the track in the MC stack first entered PHOS
+  /// \param[in] detID Index of the detector (cell) for which the hit is created
+  /// \param[in] pos Position vector of the particle first entered PHOS
+  /// \param[in] mom Momentum vector of the particle first entered PHOS
+  /// \param[in] totE Total energy of the particle entered PHOS
+  /// \param[in] time Time of the hit
+  /// \param[in] energyloss Energy deposited in this step
+  ///
+  Hit* AddHit(Int_t trackID, Int_t detID,
+                      const Point3D<float>& pos, const Vector3D<float>& mom, Double_t totE, Double_t time, Double_t eLoss) ;
+
+  ///
+  /// Register TClonesArray with hits
+  ///
+  void Register() override ;
+
+  ///
+  /// Get access to the hits
+  ///
+  std::vector<Hit>* getHits(Int_t /* iColl */) const
+  {
+     return mHits;
+  }
+
+  ///
+  /// Reset
+  /// Clean point collection
+  ///
+  void Reset() final;
+
+  ///
+  /// Steps to be carried out at the end of the event
+  /// For PHOS cleaning the hit collection and the lookup table
+  ///
+  void EndOfEvent() final;
+
+  ///
+  /// Get the PHOS geometry desciption
+  /// Will be created the first time the function is called
+  /// \return Access to the PHOS Geometry description
+  ///
+  Geometry* GetGeometry();
+
+  void updateHitTrackIndices(std::map<int, int> const& indexmapping) override {} ;
+
+ protected:
+  ///
+  /// Creating detector materials for the PHOS detector and space frame
+  ///
+  void CreateMaterials();
+
+  ///
+  /// Creating PHOS description for Geant
+  ///
+  void ConstructGeometry() override;
+
+  /// Creating PHOS/support description for Geant
+  void ConstructSupportGeometry() ;
+
+  /// Creating PHOS/calorimeter part description for Geant
+  void ConstructEMCGeometry() ;
+
+  /// Creating PHOS/CPV description for Geant
+  void ConstructCPVGeometry() ;
+
+/*
+ 
+  ///
+  /// Calculate the amount of light seen by the APD for a given track segment (charged particles only)
+  /// Calculation done according to Bricks law
+  ///
+  /// \param[in] energydeposit Energy deposited by a charged particle in the track segment
+  /// \param[in] tracklength Length of the track segment
+  /// \param[in] charge Track charge (in units of elementary charge)
+  ///
+  Double_t CalculateLightYield(Double_t energydeposit, Double_t tracklength, Int_t charge) const;
+*/
+ private:
+
+
+  //Geometry parameters
+  Bool_t mCreateCPV ;         //Should we create module with CPV
+  Bool_t mCreateHalfMod ;     //Should we create  1/2 filled module 
+  Bool_t mActiveModule[6] ;   //list of modules to create
+  Bool_t mActiveCPV[6] ;      //list of modules with CPV
+
+  //Simulation
+  Geometry * mGeom ;              //!
+  std::map<int,int> mSuperParents; //! List of tracks entered PHOS active volumes
```

Comment:
The comment `List` is a bit misleading since this is a map. Could you clarify what is being mapped to what?